tests/e2e/common/kubernetes.go

```diff
@@ -164,27 +165,55 @@ func CleanupNode(ctx context.Context, cs clientset.Interface) {
 	}
 }
 
-func CleanupNodeFeatureRules(ctx context.Context, cli *nfdclient.Clientset, namespace string) {
-	// Drop NodeFeatureRule objects
-	nfrs, err := cli.NfdV1alpha1().NodeFeatureRules().List(ctx, metav1.ListOptions{})
-	Expect(err).NotTo(HaveOccurred())
+// cleanupNodeFeatures deletes all NodeFeature objects in the given namespace
+func cleanupNodeFeatures(ctx context.Context, cli *nfdclient.Clientset, namespace string) {
+	nfs, err := cli.NfdV1alpha1().NodeFeatures(namespace).List(ctx, metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		// Omitted error, nothing to do.
+		return
+	} else {
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
```suggestion
	}
	Expect(err).NotTo(HaveOccurred())
```

Since we return we don't need the `else` block.